xls/public/c_api.cc

```diff
@@ -34,6 +34,21 @@ std::vector<std::filesystem::path> ToCpp(const char* additional_search_paths[],
 
 char* ToOwnedCString(const std::string& s) { return strdup(s.c_str()); }
 
+// Helper function that we can use to adapt to the common C API pattern when
+// we're returning an `absl::StatusOr<std::string>` value.
+bool ReturnStringHelper(absl::StatusOr<std::string>& to_return,
```

Comment:
In C we don't have parametrics, so ResultOrError is not really viable across multiple return types. Keep in mind that the purpose of this API is not to be ergonomic but to be a straightforward C ABI export of functionality -- largely we expect it'll be wrapped up for different languages in idiomatic ways. Flattening the API (i.e taking scalars or pointers-to-scalars) instead of bundling things up in aggregates is a way to try to export a flat/simple API in this case.